Update feature list in README (#2381)
serenity-rs/serenity: README.md

- **model**: Method implementations for models, acting as helper methods over
the HTTP functions.
- **standard_framework**: A standard, default implementation of the Framework
- **utils**: Utility functions for common use cases by users.
- **voice**: Enables registering a voice plugin to the client, which will handle actual voice connections from Discord.
[lavalink-rs][project:lavalink-rs] or [Songbird][project:songbird] are recommended voice plugins.
- **default_native_tls**: Default features but using `native_tls_backend`
instead of `rustls_backend`.
- **tokio_task_builder**: Enables tokio's `tracing` feature and uses `tokio::task::Builder` to spawn tasks with names if `RUSTFLAGS="--cfg tokio_unstable"` is set.
- **unstable_discord_api**: Enables features of the Discord API that do not have a stable interface. The features might not have official documentation or are subject to change.
- **simd_json**: Enables SIMD accelerated JSON parsing and rendering for API calls, use with `RUSTFLAGS="-C target-cpu=native"`
- **temp_cache**: Enables temporary caching in functions that retrieve data via the HTTP API.

> - **chrono**: Uses the `chrono` crate to represent timestamps (instead of `time`, the default)

Reviewer: ```suggestion
- **chrono**: Uses the `chrono` crate to represent timestamps. If disabled, the `time` crate is used instead.
```
This reads slightly weird, since `chrono` is a default feature.

[lavalink-rs][project:lavalink-rs] or [Songbird][project:songbird] are recommended voice plugins.
- **default_native_tls**: Default features but using `native_tls_backend`
instead of `rustls_backend`.
- **tokio_task_builder**: Enables tokio's `tracing` feature and uses `tokio::task::Builder` to spawn tasks with names if `RUSTFLAGS="--cfg tokio_unstable"` is set.
- **unstable_discord_api**: Enables features of the Discord API that do not have a stable interface. The features might not have official documentation or are subject to change.
- **simd_json**: Enables SIMD accelerated JSON parsing and rendering for API calls, use with `RUSTFLAGS="-C target-cpu=native"`
- **temp_cache**: Enables temporary caching in functions that retrieve data via the HTTP API.

> - **chrono**: Uses the chrono crate to represent timestamps
> - **time**: Uses the time crate to represent timestamps

Reviewer 2: Might be worth mentioning these are mutually exclusive.

```suggestion
- **chrono**\*: Uses the chrono crate to represent timestamps
- **time**\*: Uses the time crate to represent timestamps

....

* -- Only one of these features may be active at a time.
```

Author: Wait I just noticed the `time` feature is actually a no-op by now
![Screenshot_20230409_004707](https://user-images.githubusercontent.com/21220820/230745674-327d5430-d081-4378-b0e3-ac4dbae06b56.png)

Will remove the `time` line entirely then
